controller/controller_server/csi_controller_server.py

```diff
@@ -387,9 +387,43 @@ def ControllerUnpublishVolume(self, request, context):
 
     def ValidateVolumeCapabilities(self, request, context):
         logger.info("ValidateVolumeCapabilities")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished ValidateVolumeCapabilities")
-        return csi_pb2.ValidateVolumeCapabilitiesResponse()
+        try:
+            utils.validate_validate_volume_capabilities_request(request)
+
+            secrets = request.secrets
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+
+            space_efficiency_capabilities = {
+                key: value for key, value in request.parameters.items() if key in [
+                    config.PARAMETERS_CAPABILITIES_SPACEEFFICIENCY,
+                ]
+            }
```

Comment:
this code looks like duplication of existing code. please move it to a common function.

in addition, there are two things which I don't understand:
1. why does it use the term `capability`?
2. why does it assume that there could be more than 1?
any ideas? if not, please rename

Reply:
done